src/icmp/Icmp4.cc

```diff
@@ -132,6 +132,11 @@ Icmp4::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     icmp->icmp_cksum = CheckSum((unsigned short *) icmp, icmp_pktsize);
 
     to.getAddrInfo(S);
+    if (!S || !S->ai_addr || S->ai_family != AF_INET) {
+        debugs(42, DBG_IMPORTANT, MYNAME << " invalid destination address for ICMPv4");
```

Comment:
Is not giving a non-IPv4 address to Icmp4::SendEcho() a Squid bug rather than an I/O or similar "external" error? In other words, should not we `Assure(S->ai_family == AF_INET)` instead here (after checking that getAddrInfo() succeeded, of course)?